pkg/autoscaler/metrics/http_scrape_client_test.go

```diff
@@ -204,13 +237,61 @@ func TestHTTPScrapeClient_Scrape_ErrorCases(t *testing.T) {
 	}
 }
 
+func TestHTTPScrapeClient_ScrapeProto_ErrorCases(t *testing.T) {
+	testCases := []struct {
+		name            string
+		responseCode    int
+		responseErr     error
+		responseContext Stat
+		expectedErr     string
+	}{{
+		name:         "Non 200 return code",
+		responseCode: http.StatusForbidden,
+		expectedErr:  fmt.Sprintf(`GET request for URL %q returned HTTP status 403`, testURL),
+	}, {
+		name:         "Error got when sending request",
+		responseCode: http.StatusOK,
+		responseErr:  errors.New("upstream closed"),
+		expectedErr:  "upstream closed",
+	}}
+
+	for _, test := range testCases {
+		t.Run(test.name, func(t *testing.T) {
+			hClient := newTestHTTPClient(getProtoHTTPResponse(test.responseCode, test.responseContext, protoAcceptContent), test.responseErr)
+			sClient, err := newHTTPScrapeClient(hClient)
+			if err != nil {
+				t.Errorf("newHTTPScrapeClient=%v, want no error", err)
+			}
+			if _, err := sClient.Scrape(testURL); err != nil {
+				if !strings.Contains(err.Error(), test.expectedErr) {
+					t.Errorf("Got error message: %q, want to contain: %q", err.Error(), test.expectedErr)
+				}
+			} else {
+				t.Error("Expected error from newServiceScraperWithClient, got nil")
+			}
+		})
+	}
+}
+
 func getHTTPResponse(statusCode int, context string) *http.Response {
 	return &http.Response{
 		StatusCode: statusCode,
 		Body:       ioutil.NopCloser(bytes.NewBufferString(context)),
 	}
 }
 
+func getProtoHTTPResponse(statusCode int, context Stat, contentType string) *http.Response {
```

Comment:
```suggestion
func makeProtoResponse(statusCode int, stat Stat, contentType string) *http.Response {
```
Or `newProtoResponse`.
And I'd rename the other as well.

Reply:
I think then  it should be `makeHttpProtoResponse` because the other one cannot just be makeResponse.
It is good for readability to specify the type there.

Reply:
it should bet `makeHTTP...` but sure.